Specify the Customer Settings pages (#186)
PrestaShop/prestashop-specs: back-office/shop-parameters/customer-settings/titles.md

# **SPECIFICATIONS - CUSTOMERS SETTINGS > TITLES PAGE**

_As a merchant, I want to be able to define my customers._

## Listing

As usual, four tables options are displayed on the top-right hand corner as icons: 

- Add new<br>
- Refresh list<br>
- Show SQL query<br>
- Export to SQL manager<br>

By default, the table is divided into the following columns:

- ID<br>
- Social title<br>
- Gender<br>
- Image<br>

Searching fields are available for users to narrow down their search. 

> At the beginning of each line is a checkbox while, at the end of each line, two actions can be selected: _Edit_ and _Delete_. Checkboxes play with the 'Bulk actions' selector that allows either to 'Select all', 'Unselect all', or 'Delete selected'.

Reviewer: ```suggestion
At the beginning of each line is a checkbox while, at the end of each line, two actions can be done: _Edit_ and _Delete_. The main action is _Edit_ and it can be performed by clicking on the line.
Checkboxes play with the 'Bulk actions' selector that allows either to 'Select all', 'Unselect all', or 'Delete selected'.
```

Reviewer: We should specify the multistore behavior here:
As the add title page doesn’t have shop association block, when the user add a title it's added for all shops, no matter the context selected
So, all titles are listed here no matter the context selected (all shops, a group or a specific shop)

## Add/Edit form

Clicking 'Add new social title' and editing an existing social title redirects to a form.

**Social title.** Required field to **create or modify a social title**. Input is limited to 20 characters, otherwise an error notification is displayed: `Your entry in field %fieldname% (language %languagename%) exceeds max length %charsnumber% chars (incl. HTML tags).` in `Admin.Notifications.Error`.

If the input contains specific characters (<>;=#{}_), an error notification is displayed: `Invalid characters:` in `Admin.Global`.

> Merchants can localize the label according to the store's available languages.

Reviewer: ```suggestion
A drop-down next to the field is displayed when there are several languages installed (disabled languages are displayed in the drop-down). It allows choosing in which language the field is displayed and customizing the title according to the customer's language.
```

Only numbers should be typed in the field. If the input contains specific characters, an error notification is displayed: `Width and height must be numeric values.` in `Admin.Shopparameters.Notification`.

Leaving this field empty sets the default value.

**Image height.** Merchants can **define the image's height size**. By default, it is set to 16 pixels since PrestaShop works best with 16x16 pixels image. Images are expected to be resized if they are uploaded bigger. If merchants want to use bigger pictures, this option enables them to set the exact size required. Entering '0' makes PrestaShop use the image's original height size. A tooltip is available: `Image height in pixels. Enter "0" to use the original size.` in `Admin.Shopparameters.Help`.

Only numbers should be typed in the field. If the input contains specific characters, an error notification is displayed: `Width and height must be numeric values.` in `Admin.Shopparameters.Notification`.

Leaving this field empty sets the default value.

> 

Reviewer: ```suggestion
### Multistore
Currently, this page doesn't have shop association block. Accordingly, the title is added for all shops, no matter the context selected.

An information message should be displayed at the top of the page to inform the user of this behavior: "Note that this feature is available in all shops context only. It will be added to all your stores." (_See issue [#19415](https://github.com/PrestaShop/PrestaShop/issues/19415)_
```
We should specify the multistore behavior here:
As the add title page doesn’t have shop association block, when the user add a title it's added for all shops, no matter the context selected
By the way, an information message explaining this behavior should be displayed

Only numbers should be typed in the field. If the input contains specific characters, an error notification is displayed: `Width and height must be numeric values.` in `Admin.Shopparameters.Notification`.

Leaving this field empty sets the default value.

**Image height.** Merchants can **define the image's height size**. By default, it is set to 16 pixels since PrestaShop works best with 16x16 pixels image. Images are expected to be resized if they are uploaded bigger. If merchants want to use bigger pictures, this option enables them to set the exact size required. Entering '0' makes PrestaShop use the image's original height size. A tooltip is available: `Image height in pixels. Enter "0" to use the original size.` in `Admin.Shopparameters.Help`.

Only numbers should be typed in the field. If the input contains specific characters, an error notification is displayed: `Width and height must be numeric values.` in `Admin.Shopparameters.Notification`.

> Leaving this field empty sets the default value.

Author: ```suggestion
Leaving this field empty sets the default value.

## Multistore behavior

As the Add/Edit form does not display any shop association block, **when merchants add a social title, it is added to all shops** - no matter the context selected. So all social titles should be listed here, be it an all shops, a group, or a specific shop context.

:point_right: https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/multistoregeneralspecs.md
```

Only numbers should be typed in the field. If the input contains specific characters, an error notification is displayed: `Width and height must be numeric values.` in `Admin.Shopparameters.Notification`.

Leaving this field empty sets the default value.

**Image height.** Merchants can **define the image's height size**. By default, it is set to 16 pixels since PrestaShop works best with 16x16 pixels image. Images are expected to be resized if they are uploaded bigger. If merchants want to use bigger pictures, this option enables them to set the exact size required. Entering '0' makes PrestaShop use the image's original height size. A tooltip is available: `Image height in pixels. Enter "0" to use the original size.` in `Admin.Shopparameters.Help`.

Only numbers should be typed in the field. If the input contains specific characters, an error notification is displayed: `Width and height must be numeric values.` in `Admin.Shopparameters.Notification`.

Leaving this field empty sets the default value.

## Multistore behavior

> As the Add/Edit form does not display any shop association block, **when merchants add a social title, it is added to all shops** - no matter the context selected. So all social titles should be listed here, be it an all shops, a group, or a specific shop context.

Reviewer: ```suggestion
As the Add/Edit form does not display any shop association block, **when merchants add a social title, it is added to all shops** - no matter the context selected. 

An information message should be displayed at the top of the page to inform the user of this behavior: "Note that this feature is available in all shops context only. It will be added to all your stores." (_See issue [#19415](https://github.com/PrestaShop/PrestaShop/issues/19415)_

So all social titles should be displayed in the listing, be it an all shops, a group, or a specific shop context.
```
